olive/passes/onnx/genai_model_exporter.py

```diff
@@ -98,16 +121,49 @@ def _run_for_config(
             # snapshots that can be reused for future runs
             cache_dir = str(Path(tempfile.gettempdir()) / "hf_cache")
 
-        # currently we only support regular hf models so we can pass the name_or_path directly to model_name
-        # could also check if it is a locally saved model and pass the path to input_path but it is not necessary
+        extra_args = {"filename": str(output_model_filepath.name)}
+        if metadata_only:
+            extra_args["config_only"] = True
+            model_path = None
+            input_path = str(model.get_resource("model_path"))
+        else:
+            model_path = str(model.hf_config.model_name)
+            input_path = ""
+
         create_model(
-            model_name=str(model.model_path or model.hf_config.model_name),
-            input_path="",  # empty string for now
+            model_name=model_path,
+            input_path=input_path,
             output_dir=str(output_model_filepath.parent),
             precision=precision,
             execution_provider=target_execution_provider,
             cache_dir=cache_dir,
-            filename=str(output_model_filepath.name),
+            **extra_args,
         )
 
-        return ONNXModelHandler(output_model_filepath.parent, onnx_file_name=output_model_filepath.name)
+        # Override default search options with ones from user config
+        genai_config_filepath = str(output_model_filepath.parent / "genai_config.json")
+        with open(genai_config_filepath) as istrm:
+            genai_config = json.load(istrm)
+
+        genai_config["search"] = {**genai_config.get("search", {}), **config.get("search", {})}
```

Comment:
if both genai_config and config have search with same key, which one will take precedence?

Reply:
Values from config will override because of the order.